apm-agent-core/src/test/java/co/elastic/apm/agent/util/ExecutorUtilsTest.java

```diff
@@ -100,6 +100,5 @@ private void executeTestOnThreadPool(ThreadPoolExecutor singleThreadDaemonPool,
         }
         await().atMost(Duration.ofSeconds(10)).until(() -> !startedThread.get().isAlive());
         verify(listener).elasticThreadFinished(same(startedThread.get()));
-        verifyNoMoreInteractions(listener);
```

Comment:
Do you know what other tests are interfering? Can we find it and do whatever required?
Isn't this assertion important for us to make this test deterministic?

Reply:
OK, thanks for explaining. 
No problem removing this assertion if it is not required, but worth understanding where these thread leaks come from.
Our `co.elastic.apm.agent.impl.ElasticApmTracer#stop` implementation only shuts down the executor it manages, not others. Maybe we need to keep track on all executors and have a way to shut them all down when the tracer is stopped. Since all agent executors should be created through `ExecutorUtils`, it can potentially keep track and offer a `shutdownAllExecutors()` utility for proper cleanup.

Let's merge as is.
Based on what you observed, please decide whether it worth opening an issue to improve thread cleanup.

Reply:
I don't think we even call `stop()` properly in all tests... I would see a benefit in cleaning this up, as it might improve performance on CI/CD a little due to less memory consumption, but I don't see it as high prio.

I'll open an issue.